std/math.d

```diff
@@ -2532,74 +2532,161 @@ unittest
  *      $(TR $(TD $(NAN))            $(TD FP_ILOGBNAN) $(TD no))
  *      )
  */
-int ilogb(real x)  @trusted nothrow @nogc
+int ilogb(T)(const T x) @trusted nothrow @nogc
+    if(isFloatingPoint!T)
 {
-    version (Win64_DMD_InlineAsm)
+    import core.bitop : bsr;
+    // Provide a bsr implementation for ulong in 32-bit mode
+    int bsr_ulong(ulong x) @trusted pure nothrow @nogc
```

Comment:
Could you move this function outside from `ilogb`?
```D
static import core.bitop.bsr;
...
static if(size_t.sizeof == 4)
{
   int bsr_ulong(ulong x) @trusted pure nothrow @nogc
    {
         ...
    }
}
else
    private alias bsr_ulong = core.bitop.bsr;
```

The reason is to reduce template bloat.